### PR TITLE
feat(kit): `InputTime` add ability to use native picker on mobile

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

#1358

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/kit/components/input-time/input-time.template.html [hunk 1]
```diff
@@ -49,3 +49,17 @@
...
+<ng-template #iconContent>
+    <tui-svg
+        *polymorpheusOutlet="icon as src; context: {$implicit: size}"
+        tuiWrapper
+        appearance="icon"
+        [class.t-icon]="!computedDisabled"
+        [src]="src"
+    ></tui-svg>
+    <input
+        *ngIf="isMobile && nativePicker"
```
Comment (waterplea) [inline projects/kit/components/input-time/input-time.template.html:62]: We can also use `datalist` for items, it works for Android:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/datalist#date_and_time_types

File: projects/kit/components/input-time/input-time.module.ts [hunk 2]
```diff
@@ -4,10 +4,15 @@ import {
     TuiDataListModule,
     TuiHostedDropdownModule,
     TuiPrimitiveTextfieldModule,
+    TuiSvgModule,
     TuiTextfieldControllerModule,
+    TuiWrapperModule,
 } from '@taiga-ui/core';
+import {TuiInputTimeDirective} from '@taiga-ui/kit/components/input-time/input-time.directive';
```
Comment (waterplea) [inline projects/kit/components/input-time/input-time.module.ts:11]: Wrong paths

File: projects/kit/components/input-time/input-time.template.html [hunk 3]
```diff
@@ -49,3 +49,17 @@
         </button>
     </tui-data-list>
 </ng-template>
+
+<ng-template #iconContent>
+    <tui-svg
+        *polymorpheusOutlet="icon as src; context: {$implicit: size}"
+        tuiWrapper
+        appearance="icon"
+        [class.t-icon]="!computedDisabled"
```
Comment (waterplea) [inline projects/kit/components/input-time/input-time.template.html:58]: Is this relevant?

File: projects/kit/components/input-time/native-time/native-time.component.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import {ChangeDetectionStrategy, Component, Inject} from '@angular/core';
+import {TuiIdService, TuiMapper, TuiTime} from '@taiga-ui/cdk';
+import {TUI_TEXTFIELD_HOST} from '@taiga-ui/core';
+
+import {TuiInputTimeDirective} from '../input-time.directive';
+
+/* eslint-disable  @angular-eslint/component-max-inline-declarations */
```
Comment (splincode) [inline projects/kit/components/input-time/native-time/native-time.component.ts:7]: give me approve please ))
https://github.com/Tinkoff/taiga-ui/pull/4218

File: projects/kit/components/input-time/native-time/native-time.component.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    },
+    styleUrls: ['./native-time.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TuiNativeTimeComponent {
+    autoIdString: string;
+    constructor(
+        @Inject(TUI_TEXTFIELD_HOST) readonly host: TuiInputTimeDirective,
+        @Inject(TuiIdService)
+        readonly idService: TuiIdService,
```
Comment (waterplea) [inline projects/kit/components/input-time/native-time/native-time.component.ts:38]: private?

File: projects/kit/components/input-time/native-time/native-time.component.ts [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+    map: TuiMapper<readonly TuiTime[], string[]> = items =>
+        items.map(item => item.toString(this.host.mode));
+
+    get items(): string[] {
+        return this.host.items.map(item => item.toString(this.host.mode));
+    }
+
+    get value(): string {
+        if (this.host.value.length !== this.host.mode.length) {
```
Comment (waterplea) [inline projects/kit/components/input-time/native-time/native-time.component.ts:51]: ternary?

File: projects/kit/components/input-time/native-time/native-time.component.ts [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    autoIdString: string;
+    constructor(
+        @Inject(TUI_TEXTFIELD_HOST) readonly host: TuiInputTimeDirective,
+        @Inject(TuiIdService)
+        readonly idService: TuiIdService,
+    ) {
+        this.autoIdString = idService.generate();
+    }
+
+    map: TuiMapper<readonly TuiTime[], string[]> = items =>
```
Comment (waterplea) [inline projects/kit/components/input-time/native-time/native-time.component.ts:43]: Not used?

File: projects/kit/components/input-time/native-time/native-time.component.ts [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+        items.map(item => item.toString(this.host.mode));
+
+    get items(): string[] {
+        return this.host.items.map(item => item.toString(this.host.mode));
```
Comment (waterplea) [inline projects/kit/components/input-time/native-time/native-time.component.ts:47]: Do items really work with different modes? :)

File: projects/kit/components/input-time/native-time/native-time.component.ts [hunk 9]
```diff
@@ -0,0 +1,72 @@
+import {ChangeDetectionStrategy, Component, Inject} from '@angular/core';
+import {TuiIdService, TuiMapper, TuiTime} from '@taiga-ui/cdk';
+import {TUI_TEXTFIELD_HOST} from '@taiga-ui/core';
+
+import type {TuiInputTimeDirective} from '../input-time.directive';
+
+@Component({
+    selector: 'input[tuiTime]',
+    template: `
+        <datalist
```
Comment (waterplea) [inline projects/kit/components/input-time/native-time/native-time.component.ts:10]: I didn't know Angular allows templates for `input` self closing tags.

**Review by waterplea — COMMENTED** [review]
Change detection is not triggered upon setting value. Input updates only on next click.

File: projects/kit/components/input-time/native-time/native-time.component.ts (hunk 8 above)
Comment (vladimirpotekhin) [reply projects/kit/components/input-time/native-time/native-time.component.ts:50]: Yes, at least in chrome emulation. I added one more example to check it on android

Снимок экрана 2023-04-14 в 15 02 20

File: projects/kit/components/input-time/native-time/native-time.component.ts (hunk 9 above)
Comment (vladimirpotekhin) [reply projects/kit/components/input-time/native-time/native-time.component.ts:15]: angular makes input not self-closing this way =/
is this a legit trick or should we find another solution, what do you think?
Снимок экрана 2023-04-14 в 15 15 24

File: projects/kit/components/input-time/native-time/native-time.component.ts (hunk 8 above)
Comment (vladimirpotekhin) [reply projects/kit/components/input-time/native-time/native-time.component.ts:44]: browserstack ✅ 
Снимок экрана 2023-04-14 в 15 40 42

Comment (waterplea) [reply projects/kit/components/input-time/native-time/native-time.component.ts:44]: Awesome!

File: projects/kit/components/input-time/native-time/native-time.component.ts (hunk 9 above)
Comment (waterplea) [reply projects/kit/components/input-time/native-time/native-time.component.ts:10]: This is not valid HTML, but we only need that for Android and it works on Android in both Chrome and Firefox so I think it's ok. Makes for an awesome tweet, btw, gotta make a Stackblitz for it :)

File: projects/kit/components/input-time/input-time.component.ts [hunk 10]
```diff
@@ -103,6 +109,10 @@ export class TuiInputTimeComponent
         private readonly timeTexts$: Observable<Record<TuiTimeMode, string>>,
         @Inject(TUI_INPUT_TIME_OPTIONS)
         private readonly options: TuiInputTimeOptions,
+        @Inject(TUI_IS_MOBILE) readonly isMobile: boolean,
```
Comment (waterplea) [inline projects/kit/components/input-time/input-time.component.ts:112]: Looks like `isIos` and `isMobile` can be private.

File: projects/kit/components/input-time/native-time/native-time.component.ts [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+            ></option>
+        </datalist>
+    `,
+    host: {
+        type: 'time',
+        '[attr.list]': 'autoIdString',
+        '[tabIndex]': '-1',
+        '[value]': 'value',
+        '[step]': 'step',
+        '(change.stop.silent)': 'onChange($event.target.value)',
```
Comment (waterplea) [inline projects/kit/components/input-time/native-time/native-time.component.ts:26]: Should it be `silent`? Change detection seems to not trigger for items picked from the list.

File: projects/kit/components/input-time/native-time/native-time.component.ts [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+    },
+    styleUrls: ['./native-time.style.less'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TuiNativeTimeComponent {
+    autoIdString: string;
+    constructor(
+        @Inject(TUI_TEXTFIELD_HOST) readonly host: TuiInputTimeDirective,
+        @Inject(TuiIdService)
+        idService: TuiIdService,
```
Comment (waterplea) [inline projects/kit/components/input-time/native-time/native-time.component.ts:38]: Nit: can be on the same line as `@Inject`

**Review by nsbarsukov — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

**Pull request merged**